distributed/tests/test_client.py

```diff
@@ -1116,6 +1117,27 @@ def g():
     result = yield y._result()
     assert result == 456
 
+@gen_cluster(client=True)
+def test_upload_file_zip(c, s, a, b):
+    def g():
+        import myfile
+        return myfile.f()
+
+    with tmp_text('myfile.py', 'def f():\n    return 123') as fn_my_file, \
```

Comment:
I would really prefer this kind of test to create and use a temporary directory. You'll also need to temporarily insert the directory at the front of `sys.path`.
(you can find an example of such pattern in https://github.com/python/cpython/blob/master/Lib/test/test_import/__init__.py#L198-L215 )